### PR TITLE
[#50] script for testing transaction redeem

### PR DESCRIPTION
**Pull request by trojkat**
```
╰(clove)─λ bin/test-redeem-transaction -tx c3687b771fe495b809fbfb358fa2a393f04e31a6fae6efb52a5b737768d68d7a -p cRoFBWMvcLXrLsYFt794NRBEPUgMLf5AmnJ7VQwiEenc34z7zSpK -s 6a6277714d366a6b5a7a444a4e626546334677504b587642437a655147374d626437526a626a4c414469343964484a73415170557563654559715a7a58455152
>>> Searching for transaction: c3687b771fe495b809fbfb358fa2a393f04e31a6fae6efb52a5b737768d68d7a
>>> Found transaction: 010000000190fc6a7d655cf074532adadda4c6069c2876c414961a87653851a225e2dd74f4010000006a47304402206e8f365f6d97cb9d6b4ba53ff80740dfd57d9a9ab225513416baece4ee2c72650220688a22fd0f32e981637edca048b67cb2a49c44abccd63aab16804e532aef9d88012103142762372a0f6f2b4718cdee32fa1a3cc2465d3379312e8875ee5f9193158177000000000240420f00000000005d63a8200317ac865fe5bd4b0ffa8a5eec2f061703ee4da2548baa1ae07df3f7bf8e936d8876a9143f8870a5633e4fdac612fba47525fef082bbe96167044af7835ab17576a914812ff3e5afea281eb3dd7fce9b077e4ec6fba08b6888ac5a616f03000000001976a914812ff3e5afea281eb3dd7fce9b077e4ec6fba08b88ac00000000
>>> Transaction audit...
>>> Creating redeem transaction using secret: 6a6277714d366a6b5a7a444a4e626546334677504b587642437a655147374d626437526a626a4c414469343964484a73415170557563654559715a7a58455152
>>> Adding fee and signing...
>>> Transaction ready to be published
{'fee': 0.00075308,
 'fee_per_kb': 0.0025879,
 'fee_per_kb_text': '0.00258790 BTC / 1 kB',
 'fee_text': '0.00075308 BTC',
 'recipient_address': 'mmJtKA92Mxqfi3XdyGReza69GjhkwAcBN1',
 'size': 290,
 'size_text': '290 bytes',
 'transaction': '01000000017a8dd66877735b2ab5efe6faa6314ef093a3a28f35fbfb09b895e41f777b68c300000000ac473044022058cf77df0e150b68f5cee456eaac5a881e2ce50827a9ac70c320089c203a885e0220403ed10e56c4c628832dc67ba235d195cc579d52999d7443caa2961dd31608c0012102187b57bba6e143aca9da3557fe8bcf912379ca7ffb8c1967ca6dbde2dc695f19406a6277714d366a6b5a7a444a4e626546334677504b587642437a655147374d626437526a626a4c414469343964484a73415170557563654559715a7a58455152510000000001141c0e00000000001976a9143f8870a5633e4fdac612fba47525fef082bbe96188ac00000000',
 'transaction_hash': '612922134cb81f2bbc33c78a49cfc02c078fd1421910ef3481fee26b6ad8ec95',
 'value': 0.01,
 'value_text': '0.01000000 BTC'}
Do you want to publish this transaction (y/n): y
>>> Publishing transaction
>>> Transaction published!
>>> https://live.blockcypher.com/btc-testnet/tx/612922134cb81f2bbc33c78a49cfc02c078fd1421910ef3481fee26b6ad8ec95
```

### CODE REVIEW

**Comment by trojkat** [comment]
Something is wrong with `audit` right now.
```
╰(clove)─λ bin/test-redeem-transaction -tx bad973300c05714d28ac5dd3a8346b58f1a63675ae85cf92c69f3f45f2437ec7 -p cRoFBWMvcLXrLsYFt794NRBEPUgMLf5AmnJ7VQwiEenc34z7zSpK -s so_secret
>>> Searching for transaction: bad973300c05714d28ac5dd3a8346b58f1a63675ae85cf92c69f3f45f2437ec7
>>> Found transaction: 0100000001182f83071e2612616b3b373902b244112d84dbd62d6e6c8403290def212ce96e010000006a47304402202d45db6892ec4edba2530ba24c9aa84a9aae78b600136ab4a1c569faa0d670fe02200153b62b83c351a429bd087e392b888d3e5330dba630f5752a193c332e440ad8012103142762372a0f6f2b4718cdee32fa1a3cc2465d3379312e8875ee5f9193158177ffffffff02a0860100000000006363a8208d76283eefead8b0918fbc7eed3bbd610724c6108bac66ec86077204874fe9ef8876a9143f8870a5633e4fdac612fba47525fef082bbe961670a31353138323537363339b17576a914812ff3e5afea281eb3dd7fce9b077e4ec6fba08b6888ac2e5a6d00000000001976a914812ff3e5afea281eb3dd7fce9b077e4ec6fba08b88ac00000000
>>> Transaction audit...
Traceback (most recent call last):
  File "bin/test-redeem-transaction", line 41, in <module>
    contract = btc_network.audit_contract(tx_hex)
  File "/home/trojkat/workspace/clove/clove/network/bitcoin.py", line 431, in audit_contract
    return BitcoinContract(self, raw_transaction)
  File "/home/trojkat/workspace/clove/clove/network/bitcoin.py", line 324, in __init__
    self.locktime = datetime.fromtimestamp(self.locktime_timestamp)
OverflowError: timestamp out of range for platform time_t
```

**Comment by trojkat** [comment]
Working!

File: bin/test-redeem-transaction [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    btc_network = BitcoinTestNet()
+    wallet = btc_network.get_wallet(private_key=args.private_key)
+
+    tx_hex = get_transaction_from_address(args.transaction)
+    print('>>> Found transaction:', tx_hex)
+
+    print('>>> Transaction audit...')
+    contract = btc_network.audit_contract(tx_hex)
+    contract.show_details()
```
Comment (pafff) [inline bin/test-redeem-transaction]: It could be printed.

**Review by pafff — APPROVED** [review]

**Review by szerepak — APPROVED** [review]
:100:

**Pull request merged**